File: eBibliotekaUI/ebiblioteka_mobile/lib/models/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';
import 'knjiga.dart';
import 'korisnik.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? rezervacijaId;
  int? knjigaId;
  int? korisnikId;
  DateTime? datumRezervacije;
  DateTime? datumVracanja;
  bool? odobrena;
  Knjiga? knjiga;
  Korisnik? korisnik;

  Rezervacija(
      {this.rezervacijaId,
      this.knjigaId,
      this.korisnikId,
      this.datumRezervacije,
      this.datumVracanja,
      this.odobrena,
      this.knjiga,
      this.korisnik});

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}

File: eBiblioteka.Servisi/Database/Rezervacija.cs
using System;
using System.Collections.Generic;

namespace eBiblioteka.Servisi.Database;

public partial class Rezervacija
{
    public int RezervacijaId { get; set; }

    public int? KnjigaId { get; set; }

    public int? KorisnikId { get; set; }

    public DateTime DatumRezervacije { get; set; }

    public DateTime? DatumVracanja { get; set; }

    public bool? Odobrena { get; set; }

    public virtual Knjiga? Knjiga { get; set; }

    public virtual Korisnik? Korisnik { get; set; }
}
